File: server_integration_tests/servers/java_server/src/main/java/dev/graphlink/javaserver/impl/AuthorServiceImpl.java
package dev.graphlink.javaserver.impl;

import dev.graphlink.javaserver.generated.services.AuthorService;
import dev.graphlink.javaserver.generated.types.Author;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;

@Service
public class AuthorServiceImpl implements AuthorService {

    @Override
    public Author getAuthor(String id) {
        return Data.authors.stream().filter(a -> a.getId().equals(id)).findFirst().orElse(null);
    }

    @Override
    public List<Author> listAuthors() {
        return new ArrayList<>(Data.authors);
    }
}

File: server_integration_tests/servers/kotlin_server_blocking/src/main/kotlin/dev/graphlink/kotlinserverblocking/impl/AuthorServiceImpl.kt
package dev.graphlink.kotlinserverblocking.impl

import dev.graphlink.kotlinserverblocking.generated.services.AuthorService
import dev.graphlink.kotlinserverblocking.generated.types.Author
import org.springframework.stereotype.Service

@Service
class AuthorServiceImpl : AuthorService {
    override fun getAuthor(id: String): Author? = Data.authors.firstOrNull { it.id == id }

    override fun listAuthors(): List<Author> = Data.authors.toList()
}
